### PR TITLE
feature: implementation of the create wallet method

### PR DESCRIPTION
**Pull request by CryptologySAC**
Implementation of the create wallet method.

//TODO vanity wallets

bad idea: ~~with flexible fees in v2 it might be possible to send a 0-fee 0 ARK transaction from the new address to itself and make it known to the network that way.~~

### CODE REVIEW

**Comment by alexbarnsley** [comment]
I'm not sure if it's stated anywhere but the lowest possible fee will be 1 arktoshi.

Also, some people intentionally want cold wallets with no outgoing transactions

**Comment by faustbrian** [comment]
Fees of `0` would enable to spam the network with transactions which could be abused.

**Comment by CryptologySAC** [comment]
understood, so maybe the node could then reply with a valid response for an address that is unknown but valid

File: lib/commands/wallet/vanity.js [hunk 1]
```diff
@@ -10,13 +10,14 @@ const schema = {
   vanity: Joi.string().required()
 }
 
-function __createVanityAddress(netVersion, vanity) {
+function __createVanityAddress (netVersion, vanity) {
   const arkjs = require('arkjs')
   const bip39 = require('bip39')
-  let passphrase, address = ''
+  let passphrase
+  let address = ''
```
Comment (faustbrian) [inline lib/commands/wallet/vanity.js]: Unless you do `address += 'append to address'` you can omit the `= ''`.

Comment (CryptologySAC) [reply lib/commands/wallet/vanity.js]: thx, I imagined it would throw because null.toLowerCase().I'll update it in the next PR.

Comment (faustbrian) [reply lib/commands/wallet/vanity.js]: You can just leave things like that in, will do a refactor at the end anyways.

Comment (CryptologySAC) [reply lib/commands/wallet/vanity.js]: just got an error: Cannot read property 'toLowerCase' of undefined :)

I'll put it back in there next PR

**Pull request merged**